packages/design-system-react-native/src/components/Checkbox/Checkbox.stories.tsx

```diff
@@ -0,0 +1,85 @@
+import type { Meta, StoryObj } from '@storybook/react-native';
+import { View } from 'react-native';
+
+import { Checkbox } from './Checkbox';
+import type { CheckboxProps } from './Checkbox.types';
+
+const meta: Meta<CheckboxProps> = {
+  title: 'Components/Checkbox',
+  component: Checkbox,
+  argTypes: {
+    isSelected: {
+      control: 'boolean',
+    },
+    isDisabled: {
+      control: 'boolean',
+    },
+    isInvalid: {
+      control: 'boolean',
+    },
+    label: {
+      control: 'text',
+    },
+    twClassName: {
+      control: 'text',
+    },
+  },
+};
+
+export default meta;
+
+type Story = StoryObj<CheckboxProps>;
+
+export const Default: Story = {
+  args: {
+    isSelected: false,
+    isDisabled: false,
+    isInvalid: false,
+    label: 'Checkbox label',
+  },
+};
+
+export const isSelected: Story = {
+  render: () => (
+    <View style={{ gap: 16 }}>
+      <Checkbox isSelected={false} />
+      <Checkbox isSelected />
+    </View>
+  ),
+};
+
+export const defaultIsSelected: Story = {
+  render: () => (
+    <View style={{ gap: 16 }}>
+      <Checkbox />
+      <Checkbox defaultIsSelected />
+    </View>
+  ),
+};
+
+export const isDisabled: Story = {
+  render: () => (
+    <View style={{ gap: 16 }}>
+      <Checkbox />
+      <Checkbox isDisabled />
```

Comment:
nit: adding the label here might help document what the label looks like in disabled state
```suggestion
      <Checkbox isDisabled label="isDisabled" />
```